base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -989,7 +990,10 @@ where
         let tx_id = TxId::new_random();
         // this can be anything, so lets generate a random private key
         let pre_image = PublicKey::from_secret_key(&PrivateKey::random(&mut OsRng));
-        let hash: [u8; 32] = Sha256::digest(pre_image.as_bytes()).into();
+
+        let hash: [u8; 32] = tari_script_hash_domain()
+            .digest::<Sha256>(pre_image.as_bytes())
```

Comment:
While we're changing things, might be a good time to switch to `Blake256`, unless there's a good reason we're using SHA-2 here

Reply:
This is part of the function called `send_sha_atomic_swap_transaction` for the "SHA-XTR atomic swap transaction", so in this case I think it should remain `Sha256`.